homeassistant/components/incomfort/water_heater.py

```diff
@@ -27,16 +28,17 @@
     heater = hass.data[DOMAIN]['heater']
 
     async_add_entities([
-        IncomfortWaterHeater(client, heater)], update_before_add=True)
+        IncomfortWaterHeater(hass, client, heater)], update_before_add=True)
 
 
 class IncomfortWaterHeater(WaterHeaterDevice):
     """Representation of an InComfort/Intouch water_heater device."""
 
-    def __init__(self, client, heater):
+    def __init__(self, hass, client, heater):
```

Comment:
Don't pass in `hass`. It will be set on the entity when the entity has been added to home assistant. This means you can access `hass` with `self.hass`, after the entity has been added to home assistant.

Reply:
sorted.